File: solutions/ruby/bob/1/bob.rb
module Bob
  def self.hey(sentence)
    if sentence.strip == ''
      'Fine. Be that way!'
    elsif sentence.strip[-1] == '?'
      if sentence == sentence.upcase and sentence != sentence.downcase
        'Calm down, I know what I\'m doing!'
      else
        'Sure.'
      end
    elsif sentence == sentence.upcase and sentence != sentence.downcase
      'Whoa, chill out!'
    else
      'Whatever.'
    end
  end

end

File: solutions/csharp/bob/1/Bob.cs
public static class Bob
{
    public static string Response(string statement)
    {
        bool isQuestion = statement.Trim().EndsWith("?");
        bool isLoud = statement.Equals(statement.ToUpper()) && !(statement.Equals(statement.ToLower()));
        if (System.Text.RegularExpressions.Regex.IsMatch(statement, @"\S")) {
            if (isQuestion) {
                if (isLoud) return "Calm down, I know what I'm doing!";
                return "Sure.";
            }
            if (isLoud) return "Whoa, chill out!";
            return "Whatever.";
        }
        return "Fine. Be that way!";
    }
}